core/resp.go

```diff
@@ -228,6 +228,17 @@ func Encode(value interface{}, isSimple bool) []byte {
 			buf.Write(Encode(row.Value.Value, false))
 		}
 		return []byte(fmt.Sprintf("*%d\r\n%s", len(v), buf.Bytes()))
+	case HashMap:
+		var b []byte
+		buf := bytes.NewBuffer(b)
+		hashMap := value.(HashMap)
+		numOfKeyValuePairs := 0
+		for key, value := range hashMap {
+			buf.Write(Encode(key, false))
+			buf.Write(Encode(value, false))
+			numOfKeyValuePairs++
```

Comment:
No need to count elements this way, you can simply do a len(hashmap) to find the number of key value pairs